From 00afc190142a2c484108d9ae780d25a2e5c715ac Mon Sep 17 00:00:00 2001
From: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Date: Mon, 13 Jan 2020 15:26:23 +0530
Subject: [PATCH 0827/1851] PCI: XDMA PL PCIe: Request host bridge window
 resources

commit b2104680e7cf5d0d8919174b4d4fd0eaa215294d from
https://github.com/Xilinx/linux-xlnx.git

Request host bridge window resources so they appear in ioport_resource and
iomem_resource and are reflected in /proc/ioports and /proc/iomem.

Removing unneccessary newlines.

Signed-off-by: Bharat Kumar Gogada <bharat.kumar.gogada@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pci/controller/pcie-xdma-pl.c | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/drivers/pci/controller/pcie-xdma-pl.c b/drivers/pci/controller/pcie-xdma-pl.c
index 2120ad1a4151..d8aaf0454c3d 100644
--- a/drivers/pci/controller/pcie-xdma-pl.c
+++ b/drivers/pci/controller/pcie-xdma-pl.c
@@ -598,7 +598,6 @@ static int xilinx_pcie_probe(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 	struct pci_bus *bus;
 	struct pci_bus *child;
-
 	int err;
 	resource_size_t iobase = 0;
 	LIST_HEAD(res);
@@ -630,10 +629,15 @@ static int xilinx_pcie_probe(struct platform_device *pdev)
 		return err;
 	}
 
-	bus = pci_create_root_bus(&pdev->dev, 0,
-				  &xilinx_pcie_ops, port, &res);
-	if (!bus)
-		return -ENOMEM;
+	err = devm_request_pci_bus_resources(dev, &res);
+	if (err)
+		goto error;
+
+	bus = pci_create_root_bus(&pdev->dev, 0, &xilinx_pcie_ops, port, &res);
+	if (!bus) {
+		err = -ENOMEM;
+		goto error;
+	}
 
 	pci_scan_child_bus(bus);
 	pci_assign_unassigned_bus_resources(bus);
@@ -641,8 +645,11 @@ static int xilinx_pcie_probe(struct platform_device *pdev)
 		pcie_bus_configure_settings(child);
 	pci_bus_add_devices(bus);
 	platform_set_drvdata(pdev, port);
-
 	return 0;
+
+error:
+	pci_free_resource_list(&res);
+	return err;
 }
 
 static const struct of_device_id xilinx_pcie_of_match[] = {
-- 
2.31.1

